### PR TITLE
dpdk: 20.05 -> 21.02

### PR DESCRIPTION
**Pull request by Mic92**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 122588` at 414f54c4 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages marked as broken and skipped:</summary>
  <ul>
    <li>linuxPackages-libre.dpdk</li>
    <li>linuxPackages_hardkernel_4_14.dpdk</li>
    <li>linuxPackages_hardkernel_latest.dpdk</li>
    <li>linuxPackages_latest-libre.dpdk</li>
    <li>linuxPackages_testing_bcachefs.dpdk</li>
    <li>linuxPackages_xanmod.dpdk</li>
  </ul>
</details>
<details>
  <summary>4 packages failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/29b809cb4dac96ba86c6062408f0a5da">linuxPackages_hardened.dpdk</a></li>
    <li><a href="https://gist.github.com/4e48cdb386c1e80c732acd78a9570441">linuxPackages_latest_hardened.dpdk</a></li>
    <li><a href="https://gist.github.com/de7b263292869b50da6cde6474e8b1ff">linuxPackages_lqx.dpdk</a></li>
    <li><a href="https://gist.github.com/c02fb74af58fd4a48aedc75f4f2697c7">linuxPackages_zen.dpdk</a></li>
  </ul>
</details>
<details>
  <summary>9 packages skipped due to time constraints:</summary>
  <ul>
    <li>dpdk</li>
    <li>odp-dpdk (linuxPackages.odp-dpdk)</li>
    <li>linuxPackages.dpdk (linuxPackages_5_10.dpdk)</li>
    <li>linuxPackages_4_14.dpdk</li>
    <li>linuxPackages_4_19.dpdk</li>
    <li>linuxPackages_4_4.dpdk</li>
    <li>linuxPackages_4_9.dpdk</li>
    <li>linuxPackages_5_11.dpdk (linuxPackages_latest.dpdk)</li>
    <li>linuxPackages_5_4.dpdk</li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement. Does something look off? Please file an [issue](https://github.com/rmcgibbo/r-rmcgibbo/issues) or reach out on IRC.

File: pkgs/os-specific/linux/dpdk/default.nix [hunk 1]
```diff
@@ -1,21 +1,22 @@
 { stdenv, lib
 , kernel
 , fetchurl
+, fetchpatch
```
Comment (jonringer) [inline pkgs/os-specific/linux/dpdk/default.nix:4]: doesnt appear to be used

**Review by jonringer — COMMENTED** [review]
otherwise LGTM

```
https://github.com/NixOS/nixpkgs/pull/122588

5 packages marked as broken and skipped:
linuxPackages-libre.dpdk linuxPackages_hardkernel_4_14.dpdk linuxPackages_hardkernel_latest.dpdk linuxPackages_latest-libre.dpdk ofp

3 packages failed to build:
odp-dpdk pktgen spdk

14 packages built:
dpdk linuxPackages.dpdk linuxPackages_4_14.dpdk linuxPackages_4_19.dpdk linuxPackages_4_4.dpdk linuxPackages_4_9.dpdk linuxPackages_5_11.dpdk linuxPackages_5_4.dpdk linuxPackages_hardened.dpdk linuxPackages_latest_hardened.dpdk linuxPackages_lqx.dpdk linuxPackages_testing_bcachefs.dpdk linuxPackages_xanmod.dpdk linuxPackages_zen.dpdk
```

**Comment by Mic92** [comment]
fixed spdk build.

**Comment by Mic92** [comment]
fixed pktgen.

**Comment by Mic92** [comment]
odp-dpdk fixed.

**Comment by Mic92** [comment]
the pktgen mirror was unreliable (broken ipv6). I switched to github.

**Comment by Mic92** [comment]
Result of `nixpkgs-review pr 122588` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages marked as broken and skipped:</summary>
  <ul>
    <li>linuxPackages-libre.dpdk</li>
    <li>linuxPackages_latest-libre.dpdk</li>
    <li>ofp</li>
  </ul>
</details>
<details>
  <summary>18 packages built:</summary>
  <ul>
    <li>dpdk</li>
    <li>linuxPackages.dpdk</li>
    <li>linuxPackages.odp-dpdk</li>
    <li>linuxPackages.pktgen</li>
    <li>linuxPackages_4_14.dpdk</li>
    <li>linuxPackages_4_19.dpdk</li>
    <li>linuxPackages_4_4.dpdk</li>
    <li>linuxPackages_4_9.dpdk</li>
    <li>linuxPackages_5_11.dpdk</li>
    <li>linuxPackages_5_12.dpdk</li>
    <li>linuxPackages_5_4.dpdk</li>
    <li>linuxPackages_hardened.dpdk</li>
    <li>linuxPackages_latest_hardened.dpdk</li>
    <li>linuxPackages_lqx.dpdk</li>
    <li>linuxPackages_testing_bcachefs.dpdk</li>
    <li>linuxPackages_xanmod.dpdk</li>
    <li>linuxPackages_zen.dpdk</li>
    <li>spdk</li>
  </ul>
</details>

**Comment by Mic92** [comment]
backported to 21.05

**Pull request merged**